spec/app_spec.rb

```diff
@@ -73,7 +124,17 @@
       it 'responds 400 Bad Request' do
         expect {
           post '/polls/0/votes', { voter: 'Miyoshi', candidate: 'INVALID' }
-        }.not_to change { poll.votes.size }
+        }.not_to change { polls[0].votes.size }
+
+        expect(last_response.status).to eq 400
+      end
+    end
+
+    context 'with invalid time' do
+      it 'responds 400 Bad Request' do
+        expect {
+          post '/polls/1/votes', { voter: 'Miyoshi', candidate: 'INVALID' }
```

Comment:
candidate の値が `'INVALID'` になっているので、締切時刻の判定が壊れていても 400 が返ってテストが通ってしまいます。
きちんと締切時刻の判定をテストできるよう、それ以外の条件は正常系と一致させる必要があります。

```suggestion
          post '/polls/1/votes', { voter: 'Miyoshi', candidate: 'Alice' }
```